vulture/config.py

```diff
@@ -0,0 +1,194 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+from __future__ import print_function
+import argparse
+import sys
+from os.path import abspath, exists
+
+import toml
+
+from .version import __version__
+
+#: Possible configuration options and their respective defaults
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+NO_DEFAULT = object()
+
+
+def from_dict(data):
+    """
+    Create a new config dictionary from an existing one, assign possible
+    defaults and warn about unprocessed options.
+    """
+    # keep a copy of the keys, so we can keep track of any unprocessed
+    # values.
+    remaining_keys = set(data.keys())
```

Comment:
Done in a8a0138

For this I also needed to set default values on `argparse` arguments which did not yet have one or it would have given an error with the `NoneType`.